rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/synonyms/10_synonyms_put.yml

```diff
@@ -29,6 +29,23 @@ setup:
   - match: { reload_analyzers_details._shards.total: 0 }
   - length: { reload_analyzers_details.reload_details: 0 }
 
+---
+"Create empty synonyms set":
+  - do:
+      synonyms.put:
+        synonyms_set: test-empty-synonyms
+        body:
+          synonyms_set: []
```

Comment:
Should we support no body here, then, too?

Reply:
I'd prefer to be explicit when we want an empty synonym set. An empty body could be an error on the user side.